components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -2106,6 +2107,12 @@ private IdentityProvider getIDP(Connection dbConnection, String idPName, int idp
                 federatedIdp.setFederatedAuthenticatorConfigs(getFederatedAuthenticatorConfigs(
                         dbConnection, idPName, federatedIdp, tenantId));
 
+                // Get IdP secrets from IDN_SECRET table.
+                if (federatedIdp.getFederatedAuthenticatorConfigs().length > 0) {
+                    IdpMgtServiceComponentHolder.getInstance().getIdPSecretsProcessorService().
```

Comment:
We have to handle the case where 
```
IdpMgtServiceComponentHolder.getInstance().getIdPSecretsProcessorService()
```
returns null. Since sometimes (by misconfiguration) the component may not get activated and this may return null. There should be proper error log, not "NPE"

Reply:
Resolved with [4cca0ec](https://github.com/wso2/carbon-identity-framework/pull/4430/commits/4cca0ec26dfa11c945d145f1a0ba1ad3d2cabd20)